src/Http/RequestContext.php

```diff
@@ -118,6 +118,21 @@ public function __construct(PluginClientBuilder $builder, StreamFactoryInterface
         $this->request = $request->withBody($stream);
     }
 
+    /** @When I create the following graphql request: */
+    final public function set_the_graphql_body(string $body): void
+    {
+        $this->query = [];
+        $this->request = $this->requestFactory->createRequest('POST', '/graphql');
+
+        // let's set a default content-type
+        $this->set_the_content_type($this->getDefaultContentType());
+
+        $stream = $this->streamFactory->createStream(json_encode(['query' => $body]));
+
+        $request = $this->getRequest();
+        $this->request = $request->withBody($stream);
```

Comment:
you don't need to get the request from $this->getRequest(), as you already have it (as you are defining it). :}

```suggestion
        $this->request = $this->request->withBody($stream);
```

Reply:
It seems that psalm block me from doing this : "Cannot call method withBody on possibly null value"

Reply:
What ave you tried ? as `$this-.request` _is_ defined it can't be `null`.

Reply:
I tried again but it seem ci is broken but if you look at _/** @When I set the following body: */_ you will see that you do the same thing

Reply:
That's not the same thing. In the case of the method you are mentionning, that's because the request is created in another step. Not the same one.

While on your method, you are defining `$this->request` through the factory a few lines before.

Reply:
see https://github.com/Taluu/Behapi/pull/65/files#diff-094b631667cb05b2a145c1a02406550fR133 is defined, and called https://github.com/Taluu/Behapi/pull/65/files#diff-094b631667cb05b2a145c1a02406550fR141